apps/angular/4-typed-context-outlet/src/app/list.component.ts

```diff
@@ -25,8 +28,7 @@ import {
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class ListComponent<TItem extends object> {
-  @Input() list!: TItem[];
-
-  @ContentChild('listRef', { read: TemplateRef })
-  listTemplateRef!: TemplateRef<unknown>;
+  list = input.required<TItem[]>();
+  listTemplateRef: Signal<TemplateRef<ListTemplateContext<TItem>>> =
```

Comment:
Do you need to specify the type here? 

Personal preference: I always prefer TS to infer it